src/vulkan-renderer/wrapper/query_pool.cpp

```diff
@@ -0,0 +1,176 @@
+#include "inexor/vulkan-renderer/wrapper/query_pool.hpp"
+
+#include "inexor/vulkan-renderer/exception.hpp"
+#include "inexor/vulkan-renderer/wrapper/device.hpp"
+#include "inexor/vulkan-renderer/wrapper/make_info.hpp"
+
+#include <spdlog/spdlog.h>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+std::vector<VkQueryPipelineStatisticFlagBits> QueryPool::validate_pipeline_stats_flag_bits(
```

Comment:
I know I could make this a little more beautiful. We could have a `std::tuple` which contains the flag bit and the check it requires. Then I could loop over it and check...

Reply:
On the other hand I think it might not be worth it since it would decrease readability. Let's keep it for now.